### PR TITLE
Fix NPE when nameAttributeValue is null (#15338)

### PR DESCRIPTION
**Pull request by HyoJongPark**
This commit resolves NPEs that occur in DefaultOuth2User#getName.
If the attributes have a key, but the value comes in null, the NPE occurs, and we modified the case and added a related test case.

fixes: #15338

### CODE REVIEW

File: oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/DefaultOAuth2User.java [hunk 1]
```diff
@@ -69,8 +70,10 @@ public DefaultOAuth2User(Collection<? extends GrantedAuthority> authorities, Map
 		Assert.notEmpty(attributes, "attributes cannot be empty");
 		Assert.hasText(nameAttributeKey, "nameAttributeKey cannot be empty");
 		if (!attributes.containsKey(nameAttributeKey)) {
-			throw new IllegalArgumentException("Missing attribute '" + nameAttributeKey + "' in attributes");
+			throw new IllegalArgumentException("Missing nameAttributeKey '" + nameAttributeKey + "' in attributes");
 		}
+		Assert.notNull(attributes.get(nameAttributeKey), "Name attribute '" + nameAttributeKey + "' cannot be null");
```
Comment (jzheaux) [inline oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/user/DefaultOAuth2User.java:75]: I think `Map#get` returing `null` implies that `Map#containsKey` returns `false`.

Would you please then simply replace the `containsKey` check with the `nonNull` check?

Please change the message also to:

```
Assert.notNull(attributes.get(nameAttributeKey), "Attribute value for '" + nameAttributeKey + "' cannot be null");
```

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks! I've left some feedback inline.

Also, when you are ready, will you please squash your commits so that there is only one? Please format that final commit like so:

```
Check for null Name Attribute Value

Closes gh-15338
```

**Comment by HyoJongPark** [comment]
All review has been resolved and reformat commit.!

**Comment by jzheaux** [comment]
Thanks, @HyoJongPark! This is now merged into `main`.

**Pull request merged**